agent/agent_endpoint_test.go

```diff
@@ -5397,13 +5417,9 @@ func TestAgent_Token(t *testing.T) {
 			resp := httptest.NewRecorder()
 			req, _ := http.NewRequest(tt.method, url, tt.body)
 
-			_, err := a.srv.AgentToken(resp, req)
-			if tt.expectedErr != "" {
-				require.EqualError(t, err, tt.expectedErr)
-				return
-			}
-			require.NoError(t, err)
+			a.srv.h.ServeHTTP(resp, req)
 			require.Equal(t, tt.code, resp.Code)
+			require.Contains(t, resp.Body.String(), tt.expectedErr)
```

Comment:
Could you wrap this in an `if tt.expectedErr != ""` block? We want to compare only if it's not blank and return early (so we don't compare the fields below).

Reply:
On it! I just pushed the change.